backend/src/test/java/de/sambam/gamenighttracker/controller/UserControllerTest.java

```diff
@@ -66,15 +102,102 @@ public void getAllGamesFromDbTest() {
+                                .id("3")
+                                .sessionState("DONE")
+                                .winnerPlayerId("Sanne")
+                                .playerList(List.of(player5, player6))
+                                .build()))
                         .build()));
     }
+
+    @Test
+    @DisplayName("GET request to /user/gamesessions should return all game sessions")
+    public void getGameSessionsListTest() {
+        //WHEN
+        ResponseEntity<GameSession[]> getResponse = testRestTemplate.getForEntity(
+                "http://localhost:" + port + "user/gamesessions", GameSession[].class);
+        GameSession[] sessionList = getResponse.getBody();
+
+        //THEN
+        assertEquals(getResponse.getStatusCode(), HttpStatus.OK);
+        assertThat(sessionList.length, is(3));
+        assertThat(List.of(sessionList), containsInAnyOrder(
+                GameSession.builder()
+                        .id("1").sessionState("DONE").winnerPlayerId("Mario")
+                        .playerList(List.of(
+                                Player.builder().name("Mario").color("red").score(123)
+                                        .build(),
+                                Player.builder().name("Sanne").color("blue").score(122)
+                                        .build()))
+                        .build(),
+                GameSession.builder()
+                        .id("2")
+                        .sessionState("DONE")
+                        .winnerPlayerId("samBAM")
+                        .playerList(List.of(
+                                Player.builder().name("samBAM").color("yellow").score(100)
+                                        .build(),
+                                Player.builder().name("Mario").color("red").score(90)
+                                        .build()))
+                        .build(),
+                GameSession.builder()
+                        .id("3")
+                        .sessionState("DONE")
+                        .winnerPlayerId("Sanne")
+                        .playerList(List.of(
+                                Player.builder().name("Sanne").color("blue").score(80)
+                                        .build(),
+                                Player.builder().name("samBAM").color("yellow").score(70)
+                                        .build()))
+                        .build()));
+
+    }
+
+    @Test
+    @DisplayName("getPlayerList() should return all players from Db w/o duplicates")
+    public void getPlayerListTest() {
+        //WHEN
+        ResponseEntity<PlayerDto[]> getResponse = testRestTemplate.getForEntity(
+                "http://localhost:" + port + "user/players", PlayerDto[].class);
+        PlayerDto[] playerDtoList = getResponse.getBody();
+
+        //THEN
+        assertTrue(getResponse.getStatusCode().equals(HttpStatus.OK));
+        assertThat(getResponse.getBody().length, is(3));
```

Comment:
not necessarily needed because your next assert checks whether your list includes all expected objects